### PR TITLE
simplifies creation of routing_node -- issue #379

### PR DESCRIPTION
**Pull request by mmoadeli**
fixes issue #379

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/380)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
reviewing

**Comment by benjaminbollen** [comment]
First comment: this will change the public API to Vaults

File: src/routing_node.rs [hunk 1]
```diff
@@ -216,7 +211,7 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
             Some(relocated_name) => {
                 self.id.assign_relocated_name(relocated_name);
                 debug_assert!(self.id.is_relocated());
-                let mut membrane = RoutingMembrane::<F>::new(
+                let mut membrane = RoutingMembrane::<G::InterfaceType>::new(
```
Comment (benjaminbollen) [inline src/routing_node.rs]: This is a nice twist

**Comment by benjaminbollen** [comment]
Paused until version 0.2.0

**Comment by benjaminbollen** [comment]
This breaks the public API for Vaults; please revert or add as a feature or link the fix in Vaults as Asia is running late iin the day

**Comment by benjaminbollen** [comment]
Correction: it got merged as part of the #383 #372 ; but there it also contains the revert-commit;

so @mmoadeli please keep your own branch of this work and for the next version make the PR again.

**Pull request merged**